pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -45,6 +45,16 @@ type PodInterface struct{}
 
 func (l *PodInterface) Unplug() {}
 
+func findPodInterface(ifaces []api.Interface) (int, error) {
+	for i, iface := range ifaces {
+		// TODO:(ihar) find a more robust way to detect pod interface
```

Comment:
You mean "Pod" set to non-nil, right? If that's what you mean, I don't think this applies here because the api.Network type is not the same as v1.Network and doesn't have anything specific to indicate whether it's a pod network. This module is in virt-launcher and not a direct representation of VirtualMachineSpec.

There is probably a better way of doing it, but it would require somehow passing additional information about in-spec network definitions related to domain networks from api layer.

Reply:
Yes I see. The proper way is with libvirt 3.9 with this fix: https://bugzilla.redhat.com/show_bug.cgi?id=1434451. One of the reasons why we have a mandatory name field on all devices is that we can set this name as qemu alias. You would then search for the device name like described above on the vm. Then you take the name and search for the interface in the libvirt struct where this name is set as device alias.